programs/mango-v4/tests/cases/test_health_compute.rs

```diff
@@ -142,6 +169,145 @@ async fn test_health_compute_tokens_during_maint_weight_shift() -> Result<(), Tr
     Ok(())
 }
 
+// Try to reach compute limits in health checks by having many different tokens in an account and using fallback oracles for them
+#[tokio::test]
+async fn test_health_compute_tokens_fallback_oracles() -> Result<(), TransportError> {
+    let mut test_builder = TestContextBuilder::new();
+    test_builder.test().set_compute_max_units(450_000);
+    let context = test_builder.start_default().await;
+    let solana = &context.solana.clone();
+
+    let num_tokens = 8;
+
+    let admin = TestKeypair::new();
+    let owner = context.users[0].key;
+    let payer = context.users[1].key;
+    let mints = &context.mints[0..num_tokens];
+
+    let mut fallback_oracle_kps = Vec::with_capacity(num_tokens);
+    for _ in 0..num_tokens {
+        fallback_oracle_kps.push(TestKeypair::new());
+    }
+    let fallback_metas: Vec<AccountMeta> = fallback_oracle_kps
+        .iter()
+        .map(|x| AccountMeta {
+            pubkey: x.pubkey(),
+            is_signer: false,
+            is_writable: false,
+        })
+        .collect();
+
+    // let fallback_metas = vec![];
+
+    //
+    // SETUP: Create a group and an account
+    //
+
+    let GroupWithTokens { group, tokens, .. } = GroupWithTokensConfig {
+        admin,
+        payer,
+        mints: mints.to_vec(),
+        ..GroupWithTokensConfig::default()
+    }
+    .create(solana)
+    .await;
+
+    let account =
+        create_funded_account(&solana, group, owner, 0, &context.users[1], &[], 1000, 0).await;
+
+    let mut cu_measurements = vec![];
+    for token_account in &context.users[0].token_accounts[..mints.len()] {
+        deposit_cu_datapoint(solana, account, owner, *token_account).await;
+    }
```

Comment:
might be leftover code?

Reply:
Maybe I'm being too cautious here but I was trying to isolate the CU of only a stale oracle deposit. I didn't know if creating a token position would overestimate the CU usage of a fallback..

Reply:
ah, ok!